core/dbt/events/functions.py

```diff
@@ -220,6 +215,15 @@ def send_to_logger(l: Union[Logger, logbook.Logger], level_tag: str, log_line: s
         )
 
 
+def warn_or_error(event, node=None):
+    if flags.WARN_ERROR:
+        from dbt.exceptions import raise_compiler_error
+
+        raise_compiler_error(scrub_secrets(event.info.msg, env_secrets()), node)
```

Comment:
I know this code has been moved and isn't new, but should we perhaps fire the event on both branches? Otherwise, I would guess the exception won't be logged.

Reply:
We comfy with importing the exception down yonder inline rather than at the top? Do we have a style convention here?

Reply:
Doing it here actually prevents a circular dependency.  My next task #5958 should help resolve this to a better solution.